cli_tools/gce_vm_image_import/args/args.go

```diff
@@ -0,0 +1,289 @@
+//  Copyright 2020 Google Inc. All Rights Reserved.
+//
+//  Licensed under the Apache License, Version 2.0 (the "License");
+//  you may not use this file except in compliance with the License.
+//  You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+//  Unless required by applicable law or agreed to in writing, software
+//  distributed under the License is distributed on an "AS IS" BASIS,
+//  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//  See the License for the specific language governing permissions and
+//  limitations under the License.
+
+package args
+
+import (
+	"flag"
+	"fmt"
+	"io/ioutil"
+	"os"
+	"strings"
+	"time"
+
+	daisy_utils "github.com/GoogleCloudPlatform/compute-image-tools/cli_tools/common/utils/daisy"
+	"github.com/GoogleCloudPlatform/compute-image-tools/cli_tools/common/utils/param"
+	"github.com/GoogleCloudPlatform/compute-image-tools/cli_tools/gce_vm_image_import/importer"
+)
+
+// Flags that are validated.
+const (
+	imageFlag          = "image_name"
+	clientFlag         = "client_id"
+	dataDiskFlag       = "data_disk"
+	osFlag             = "os"
+	customWorkflowFlag = "custom_translate_workflow"
+)
+
+// ParsedArguments holds the structured results of parsing CLI arguments.
+type ParsedArguments struct {
+	Img         importer.ImageSpec
+	Env         importer.Environment
+	Translation importer.TranslationSpec
+}
+
+// ParseArgs parses, validates, and populates the CLI arguments that are
+// used by the importer tool.
+func ParseArgs(args []string, populator param.Populator, sf importer.SourceFactory) (ParsedArguments, error) {
+	flagSet := flag.NewFlagSet("image-import", flag.ContinueOnError)
+	// Don't write parse errors to stdout, instead propagate them via an
+	// exception since we use flag.ContinueOnError.
+	flagSet.SetOutput(ioutil.Discard)
+
+	img := importer.ImageSpec{}
+	env := importer.Environment{
+		CurrentExecutablePath: os.Args[0],
+	}
+	translation := importer.TranslationSpec{}
+
+	registerImageSpec(flagSet, &img)
+	registerEnvironment(flagSet, &env)
+	registerTranslationSpec(flagSet, &translation)
```

Comment:
I used `register` since we need to register where the flag parser will store its result.  `populate` is probably a misnomer since we use `populate` elsewhere to mean backfilling omitted parameters.

How about: `associateImageSpec`?

Reply:
or registerFlagsForImageSpec